src/main/java/net/mcreator/workspace/elements/VariableElementType.java

```diff
@@ -22,55 +22,42 @@
 import net.mcreator.generator.mapping.NameMapper;
 import net.mcreator.workspace.Workspace;
 
-public enum VariableElementType {
+import java.awt.*;
 
-	@SerializedName("string") STRING(0x609986), @SerializedName("logic") LOGIC(
-			0x607c99), @SerializedName("number") NUMBER(0x606999), @SerializedName("itemstack") ITEMSTACK(0x996069);
+public class VariableElementType {
 
-	private final int color;
-
-	VariableElementType(int color) {
-		this.color = color;
-	}
+	private String name;
+	private int color;
+	private String defaultValue;
+	private String blocklyVariableType;
+	private String javaClass;
 
 	public int getColor() {
 		return color;
 	}
 
-	public String toDependencyType() {
-		switch (this) {
-		case NUMBER:
-			return "number";
-		case LOGIC:
-			return "boolean";
-		case STRING:
-			return "string";
-		case ITEMSTACK:
-			return "itemstack";
-		}
+	public String getName() {
+		return name;
+	}
+
+	public String getBlocklyVariableType() {
+		return blocklyVariableType;
+	}
 
-		return null;
+	public String getJavaClass() {

```

Comment:
Looking again this may not be even needed as you already have `getJavaType`, or is this one different. I think `getJavaType` could be used in all cases as it also uses mapping system

Reply:
You're right. The `getJavaType` method using mappings is the only one needed.